app/stac_api/serializers.py

```diff
@@ -338,13 +350,61 @@ def validate_type(self, value):
         '''
         return normalize_and_validate_media_type(value)
 
+    def _validate_href_url(self, url):
+        # thanks https://stackoverflow.com/questions/3809401/
+        # what-is-a-good-regular-expression-to-match-a-url
+        url_regex = (
+            r'https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}'
+            r'\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)'
+        )
+
+        if not re.match(url_regex, url):
+            errors = {'href': _('Invalid URL provided')}
+            raise serializers.ValidationError(code='payload', detail=errors)
+
+        # get the collection regex. if there's none, allow all urls
+        collection_regex = self.collection.external_asset_pattern or r".*"
+
+        if not re.search(collection_regex, url):
+            errors = {'href': _("Invalid URL provided. It doesn't match the collection pattern")}
+            raise serializers.ValidationError(code='payload', detail=errors)
+
+        try:
+            response = requests.head(url, timeout=settings.EXTERNAL_URL_REACHABLE_TIMEOUT)
```

Comment:
an alternative for this check would be to just request a small range of the response (similar to e.g. `curl <URL> -i -H "Range: bytes=0-511"`, but this would require the external server to support `range`s, dunno if it makes sense to have that requirement (for the MCH case it probably makes sense, since they have big files as well where support of `range`s would allow for pause/resume download etc. 
I'll communicate to MCH that we'll do the check with `head`, mentioning the alternative using `range`.

Reply:
According to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Range the servers that don't support the header *may* ignore it. Don't know though what the common case is.

What would be a benefit of the Range over the Head?

Reply:
Ok, good to know! I'm gonna leave it for now then, and we now have much better logging, too!